homeassistant/components/bitvavo/sensor.py

```diff
@@ -0,0 +1,279 @@
+import logging
+
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ASSET_VALUE_BASE,
+    ASSET_VALUE_CURRENCIES,
+    CONF_ASSET_TICKERS,
+    CONF_BALANCES,
+    CONF_OPEN_ORDERS,
+    CONF_TICKERS,
+    CURRENCY_ICONS,
+    DEFAULT_COIN_ICON,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Bitvavo sensors."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    entities = []
+
+    for market in coordinator.data[CONF_TICKERS]:
+        entities.append(Ticker(coordinator, market))
+
+    if CONF_BALANCES in coordinator.data:
+        for balance in coordinator.data[CONF_BALANCES]:
+            if coordinator.data[CONF_BALANCES][balance]["available"] > "0":
+                entities.append(Balance(coordinator, balance))
+
+        for currency in ASSET_VALUE_CURRENCIES:
+            entities.append(TotalAssetValue(coordinator, currency))
+
+    if CONF_OPEN_ORDERS in coordinator.data:
+        entities.append(OpenOrder(coordinator, coordinator.data[CONF_OPEN_ORDERS]))
+
+    async_add_entities(entities, False)
+
+
+class Ticker(CoordinatorEntity):
+    """Implementation of the ticker sensor."""
+
+    def __init__(self, coordinator, symbol):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._symbol = symbol
+
+        self._name = f"Bitvavo Ticker - {self._symbol}"
+        self._unique_id = f"bitvavo_ticker_{self._symbol})"
+
+    def _get_data_property(self, property_name):
+        """Return the property from self.coordinator.data."""
+        return self.coordinator.data[CONF_TICKERS][self._symbol][property_name]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return round(float(self._get_data_property("price")), 4)
+
+    @property
+    def unique_id(self):
+        """Return a unique id for the sensor."""
+        return self._unique_id
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._get_data_property("quote")
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return CURRENCY_ICONS.get(self.unit_of_measurement, DEFAULT_COIN_ICON)
+
+    @property
+    def device_state_attributes(self):
+        """Return additional sensor state attributes."""
+        return {
+            "symbol": self._symbol,
+            "bid_price": self._get_data_property("bid"),
+            "ask_price": self._get_data_property("ask"),
+            "bid_size": self._get_data_property("bidSize"),
+            "ask_size": self._get_data_property("askSize"),
+            "currency": self._get_data_property("base"),
+            "quote_asset": self._get_data_property("quote"),
```

Comment:
Shouldn't these be separate sensors?

Reply:
This is information only belonging to this sensor, isn't practical to separate them.